httpclient/sentryhttpclient.go

```diff
@@ -0,0 +1,140 @@
+	if originalRoundTripper == nil {
+		originalRoundTripper = http.DefaultTransport
+	}
+
+	// Configure trace propagation targets
+	var tracePropagationTargets []string
+	if hub := sentry.CurrentHub(); hub != nil {
+		client := hub.Client()
+		if client != nil {
+			clientOptions := client.Options()
+			if clientOptions.TracePropagationTargets != nil {
+				tracePropagationTargets = clientOptions.TracePropagationTargets
+			}
+
+		}
+	}
+
+	t := &SentryRoundTripper{
+		originalRoundTripper:    originalRoundTripper,
+		tracePropagationTargets: tracePropagationTargets,
+	}
+
+	for _, opt := range opts {
+		if opt != nil {
+			opt(t)
+		}
+	}
+
+	return t
+}
+
+// SentryRoundTripper provides a http.RoundTripper implementation for Sentry Requests module.
+type SentryRoundTripper struct {
+	originalRoundTripper http.RoundTripper
+
+	tracePropagationTargets []string
+}
+
+func (s *SentryRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	// Respect trace propagation targets
+	if len(s.tracePropagationTargets) > 0 {
+		requestURL := request.URL.String()
+		foundMatch := false
+		for _, target := range s.tracePropagationTargets {
+			if strings.Contains(requestURL, target) {
+				foundMatch = true
+				break
+			}
+		}
+
+		if !foundMatch {
+			return s.originalRoundTripper.RoundTrip(request)
+		}
+	}
+
+	// Only create the `http.client` span only if there is a parent span.
+	parentSpan := sentry.SpanFromContext(request.Context())
+	if parentSpan == nil {
+		if hub := sentry.GetHubFromContext(request.Context()); hub != nil {
+			request.Header.Add("Baggage", hub.GetBaggage())
+			request.Header.Add("Sentry-Trace", hub.GetTraceparent())
+		}
+
+		return s.originalRoundTripper.RoundTrip(request)
+	}
+
+	cleanRequestURL := request.URL.Redacted()
+
+	span := parentSpan.StartChild("http.client", sentry.WithTransactionName(fmt.Sprintf("%s %s", request.Method, cleanRequestURL)))
+	defer span.Finish()
+
+	span.SetData("http.query", request.URL.Query().Encode())
+	span.SetData("http.fragment", request.URL.Fragment)
+	span.SetData("http.request.method", request.Method)
+	span.SetData("server.address", request.URL.Hostname())
+	span.SetData("server.port", request.URL.Port())
+
+	// Always add `Baggage` and `Sentry-Trace` headers.
+	request.Header.Add("Baggage", span.ToBaggage())
+	request.Header.Add("Sentry-Trace", span.ToSentryTrace())
+
+	response, err := s.originalRoundTripper.RoundTrip(request)
+	if err != nil {
+		span.Status = sentry.SpanStatusInternalError
+	}
+
+	if response != nil {
+		span.Status = sentry.HTTPtoSpanStatus(response.StatusCode)
+		span.SetData("http.response.status_code", response.StatusCode)
+		span.SetData("http.response_content_length", response.ContentLength)
+	}
+
+	return response, err
+}
+
+// SentryHTTPClient provides a default HTTP client with SentryRoundTripper included.
+// This can be used directly to perform HTTP request.
+var SentryHTTPClient = &http.Client{
```

Comment:
nit: can we change the default name here? it's already too verbose and the package already specifies that we are referring to the sentryhttpclient.
```suggestion
var Client = &http.Client{
```